src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -509,7 +509,7 @@ def __init__(
         # compute macro weights
         df = pandas.DataFrame(data=evaluation_triples.numpy(), columns=list(self.COLUMNS))
         self.precomputed_weights = dict()
-        self.weights = {}
+        self.weights = defaultdict(list)
```

Comment:
i'm worried that using a defaultdict is a bandage for properly maintaining this dictionary

Reply:
can we initialize the dict with the appropriate keys pointing to empty lists then figure out after where it gets cleared and reset?

Reply:
Using `defaultdict`s is the same way other buffers are handled (e.g., `ranks`)

Reply:
```suggestion
        self.weights = {}
```